src/main/java/hudson/remoting/Launcher.java

```diff
@@ -127,17 +121,25 @@ public void setTextMode(boolean b) {
         System.out.println("Running in "+mode.name().toLowerCase(Locale.ENGLISH)+" mode");
     }
 
+    /**
+     * @deprecated use {@link #secret}, {@link #name}, {@link #urls}, {@link #webSocket}, {@link #tunnel},
+     * {@link #workDir}, {@link #internalDir}, and/or {@link #failIfWorkDirIsMissing} directly.
+     */
     @Option(name="-jnlpUrl",usage="instead of talking to the controller via stdin/stdout, " +
             "emulate a JNLP client by making a TCP connection to the controller. " +
-            "Connection parameters are obtained by parsing the JNLP file.")
+            "Connection parameters are obtained by parsing the JNLP file.", forbids = {"-direct", "-name", "-tunnel", "-url", "-webSocket"})
+    @Deprecated
     public URL agentJnlpURL = null;
```

Comment:
The most complicated mode of this plugin is when `-jnlpUrl` is passed in, which is not done by the Docker inbound agent image used for the Kubernetes plugin but which is recommended when creating an inbound agent in the UI. In this mode, you have to pass some arguments in the command-line invocation, but some other arguments can be provided by the server from the pseudo-JNLP endpoint. This is quite complex and it seems simpler to prefer that all clients simply invoke the tool with the parameters they want rather than relying on the pseudo-JNLP endpoint as a primitive bootstrapping mechanism.